crates/oxc_linter/src/rules/eslint/no_unused_vars/usage.rs

```diff
@@ -847,7 +847,15 @@ impl<'a> Symbol<'_, 'a> {
         loop {
             node = match node.kind() {
                 AstKind::TSTypeQuery(_) => return true,
-                AstKind::TSQualifiedName(_) | AstKind::TSTypeName(_) => {
+                // TODO, this was TSTypeName, this is probably wrong
+                AstKind::IdentifierReference(_) => {
+                    let Some(parent) = self.nodes().parent_node(node.id()) else {
+                        debug_assert!(false);
+                        return false;
+                    };
+                    parent
+                }
+                AstKind::TSQualifiedName(_) => {
```

Comment:
we can drop the `TODO` comment. this is correct. ( i think we can also collapse these two amtch arms?

tis test is trying to work out wehter the refernce is used as
`type X`
as the argument to `TSTypeQuery` could be an `IdentifierReference` OR `TSQualifiedName`, we need to go up with both of them